[FerdiHS] iP

FerdiHS opened the pull request:
# Not A Chatbot
> "Deal with difficult task while they are easy. Act on large issues while they are small." ~ Lao Tzu ([source](https://www.google.com/url?sa=i&url=https%3A%2F%2Fquotesgram.com%2Fquotes-about-big-tasks%2F&psig=AOvVaw2tW-zeWMiGWhhlll_jtN-2&ust=1693932325084000&source=images&cd=vfe&opi=89978449&ved=0CA4QjRxqFwoTCMDK1OezkYEDFQAAAAAdAAAAABAP))

What is it?
* Actually a chat bot
* Text-based program
* Help you manage your task
* It is **FREE**

What you need to do to try it?
- [ ] Go to this [link](https://github.com/FerdiHS/ip/releases).
- [ ] Download the .jar file of the latest release.
- [ ] Run the code with your local directory.
- [ ] The program is ready to be used.

Features:
1. Differentiates tasks into todos, deadlines, and events.
2. Adds tasks.➕
3. Stores tasks
4. Shows the list of tasks.
5. Marks tasks as completed. ✅
6. Unmarks tasks as not completed. ❎
7. Removes tasks. ➖
8. Finds tasks🔎


Here's the example of the code :
```java
public static void main(String[] args) {
        new Duke(DIRECTORY_NAME, FILE_NAME).run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

FerdiHS committed:
Level-0

FerdiHS committed:
Level-1

FerdiHS committed:
Level-2

FerdiHS committed:
Level-3

FerdiHS committed:
Level-3

FerdiHS committed:
Level-3

FerdiHS committed:
Level-4

FerdiHS committed:
A-TextUiTesting

FerdiHS committed:
Level-5

FerdiHS committed:
Level-6

FerdiHS committed:
Level-6

FerdiHS committed:
Add storage

FerdiHS committed:
Merge pull request #1 from FerdiHS/branch-Level-7

Merge branch-level-7 to master

FerdiHS committed:
Use Java LocalDate

FerdiHS committed:
Merge pull request #2 from FerdiHS/branch-Level-8

Merge branch-Level-8 to master

FerdiHS committed:
Add TaskList

FerdiHS committed:
Add Ui class

FerdiHS committed:
Add Parser class

FerdiHS committed:
Add Parser and Command class

FerdiHS committed:
Remove some unused methods and variables

FerdiHS committed:
Merge pull request #3 from FerdiHS/A-MoreOOP

Merge branch A-MoreOOP to master

FerdiHS committed:
Add duke package

FerdiHS committed:
Merge pull request #4 from FerdiHS/add-gradle-support

Merge branch add-gradle-support

FerdiHS committed:
Add JUnit Testing

FerdiHS committed:
Update the directory of the storage for JUnit Test

FerdiHS committed:
Fix code styling

FerdiHS committed:
Merge pull request #5 from FerdiHS/branch-A-CodingStandard

Merge branch-A-CodingStandard to master

FerdiHS committed:
Add and fix JavaDoc

FerdiHS committed:
Merge branch 'master' into branch-A-JavaDoc

FerdiHS committed:
Merge pull request #6 from FerdiHS/branch-A-JavaDoc

Merge branch-A-JavaDoc to master

FerdiHS committed:
Fix minor mistake caused by merging

FerdiHS committed:
Add find feature

FerdiHS committed:
Merge branch 'master' into branch-Level-9

FerdiHS committed:
Merge pull request #7 from FerdiHS/branch-Level-9

Merge branch-Level-9 to master

FerdiHS committed:
Fix the style with the help of check style

FerdiHS committed:
Merge pull request #8 from FerdiHS/branch-A-CheckStyle

Merge branch-A-CheckStyle to master

FerdiHS committed:
Add temporary GUI

FerdiHS committed:
Add 'bye' option to close the program

FerdiHS committed:
Merge pull request #9 from FerdiHS/branch-Level-10

Merge branch-Level-10 to master

teozhengyang commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,57 @@
+package duke.task;
+import java.time.LocalDate;
```
Might be better to leave a space between the package and import statements since you did it for your other files for consistency reasons.

teozhengyang commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,68 @@
+package duke.task;
+import java.time.LocalDate;
```
Might be better to leave a space between the package and import statements since you did it for your other files for consistency reasons.

teozhengyang commented on src/main/java/duke/task/ToDo.java:
```diff
@@ -0,0 +1,37 @@
+package duke.task;
+import duke.exception.InvalidToDoException;
```
Might be better to leave a space between the package and import statements since you did it for your other files for consistency reasons.

teozhengyang reviewed (commented):
Overall LGTM. Followed all the coding standards.

SinhaVedant commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,72 @@
+    static final String FILE_NAME = "storage.txt";
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
```
I like that you kept the variable name tasks plural.

SinhaVedant commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,59 @@
+     * @return the response
+     */
+    public static String parse(String message, Ui ui, TaskList tasks, Storage storage) {
+        try {
+            String messageType = message.split(" ")[0];
+            if (messageType.equals("list")) {
+                return Command.list(message, ui, tasks, storage);
+            } else if (messageType.equals("mark")) {
+                return Command.mark(message, ui, tasks, storage);
+            } else if (messageType.equals("unmark")) {
+                return Command.unmark(message, ui, tasks, storage);
+            } else if (messageType.equals("todo")) {
+                return Command.addToDo(message, ui, tasks, storage);
+            } else if (messageType.equals("deadline")) {
+                return Command.addDeadline(message, ui, tasks, storage);
+            } else if (messageType.equals("event")) {
+                return Command.addEvent(message, ui, tasks, storage);
+            } else if (messageType.equals("delete")) {
+                return Command.delete(message, ui, tasks, storage);
+            } else if (messageType.equals("find")) {
+                return Command.find(message, ui, tasks, storage);
+            } else if (message.equals("bye")) {
+                return ui.printEnd();
+            } else {
+                throw new InvalidCommandException();
+            }
```
I like that you avoided deep-nesting here.

SinhaVedant commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,127 @@
+package duke.storage;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.ToDo;
```
I like how you ordered your import statements.

SinhaVedant commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,127 @@
+     * @throws IOException if the task read is invalid
+     */
+    public static Task readTask(String taskStoredInString) throws IOException {
+        String[] taskInStringSplit = taskStoredInString.split(" \\| ");
```
Perhaps the name of variable `taskInStringSplit` could be improved by reordering to `splitTaskInString`?

SinhaVedant reviewed (commented):
LGTM. Just a few nits to fix.

FerdiHS committed:
Fix build.gradle to enable running from graddle

FerdiHS committed:
Add assert

The current program do not have any assertion checks, making it
challenging to ensure the correctness of the assumption made in the application.

Assert reassures the assumption made in the program.

Adding assert to some methods to reassures assumption made. For example,
in the method list in Command, assertis used to reassure if the first
word of the message is "list".

Using Java assert statement is an efficient way to perform runtime
checks on the program which allow catching unexpected conditions and
failures early in the development process.

FerdiHS committed:
Improve code quality

FerdiHS committed:
Merge pull request #10 from FerdiHS/branch-A-Assertions

Add assert

FerdiHS committed:
Merge branch 'master' into branch-A-CodeQuality

FerdiHS committed:
Merge pull request #11 from FerdiHS/branch-A-CodeQuality

Improve code quality

FerdiHS committed:
Remove readTask method in Storage.java since it is no longer needed

FerdiHS committed:
Add TimedTask feature

Before there was not any feature to make and keep track of task which
can be done in a certain period of time without certainty of the start
and the end of the task

Enabling to add, mark, unmark, and remove TimedTask which is done in the
same way as the other type of tasks.

This way is used to maintain the uniform of code with the other type of
tasks.

FerdiHS committed:
Merge pull request #13 from FerdiHS/branch-BCD-Extension

Merge branch-BCD-Extension to master

FerdiHS committed:
Add background color to the chat box

FerdiHS committed:
Merge pull request #14 from FerdiHS/branch-A-BetterGui

Add background color to the chat box

FerdiHS committed:
Change bot name from Duke to Boti

FerdiHS committed:
Change some behaviour and response of the chat bot

FerdiHS committed:
Merge pull request #15 from FerdiHS/branch-A-Personality

Merge branch-A-Personality to master

FerdiHS committed:
Makes minor changes in the chatbot response messages

FerdiHS committed:
Make further minor changes in the response of the chatbot

FerdiHS committed:
Add User Guide

FerdiHS committed:
Make and handle more command exceptions

FerdiHS committed:
Merge pull request #16 from FerdiHS/branch-A-MoreErrorHandling

Merge branch-A-MoreErrorHandling to master

FerdiHS committed:
Add more unit test

FerdiHS committed:
Add table of content to the User Guide

FerdiHS committed:
Update build.graddle
